examples/plot_raster_stack_crop.py

```diff
@@ -0,0 +1,211 @@
+# and either export them together a single stacked .tif file or work with them in Python
+# directly as a stacked numpy array.
+#
+# To begin using the EarthPy stack function, import the needed packages
+# and create an array to be plotted. Below we plot the data as continuous with a colorbar
+# using the ``plot_bands()`` function.
+
+###############################################################################
+# Import Packages
+# ------------------------------
+#
+# You will need several packages to stack your raster. Geopandas will allow you to quickly
+# open up a shapefile that will be used later to crop your data. You will primarily be
+# using the spatial module
+# of EarthPy, but other packages will be needed as well.
+
+import os
+from glob import glob
+import matplotlib.pyplot as plt
+import rasterio as rio
+from rasterio.plot import plotting_extent
+import earthpy as et
+import geopandas as gpd
+import earthpy.spatial as es
+import earthpy.plot as ep
+
+
+###############################################################################
+# Get Example Data Ready for Stack
+# ----------------------------------
+# With EarthPy we can create a stack from all of the Landsat .tif files (one per band)
+# in a folder with the `es.stack()` function.
+
+#############################################################################
+# Error found on Windows systems
+# -------------------------------
+# .. note::
+#       If you are running this script on a Windows system, there is a
+#       known bug with ``.to_crs()``, which is used in this script. If an error
+#       occurs, you have to reset your os environment with the command
+#       ```os.environ["PROJ_LIB"] = r"path-to-share-folder-in-environment"```.
+
+# Getting sample data from EarthPy and setting your home working directory
+
+data_path_1 = et.data.get_data("cs-test-landsat")
+os.chdir(os.path.join(et.io.HOME, "earth-analytics"))
+
+# Preparing the landsat bands to be stacked using glob and sort
+
+landsat_bands_data_path = "data/cs-test-landsat/LC08_L1TP_034032_20160621_20170221_01_T1_sr_band*[2-4]*.tif"
+stack_band_paths = glob(landsat_bands_data_path)
+stack_band_paths.sort()
+
+# Creating output directory and the output path
+
+output_dir = os.path.join("data", "outputs")
+if os.path.isdir(output_dir) == False:
+    os.mkdir(output_dir)
+
+raster_out_path = os.path.join(output_dir, "raster.tiff")
+
+####################################################################################
+# Stacking the prepared bands
+# ---------------------------
+# The stack function has an optional output argument, where you can write the raster
+# to a tiff file in a folder. If you want to use this functionality, make sure there
+# is a folder to write your tiff file to.
+# The Stack function also returns two object, an array and a RasterIO profile. Make
+# sure to be catch both in variables.
+
+# Stacking the Landsat bands
+
+os.chdir(os.path.join(et.io.HOME, "earth-analytics"))
```

Comment:
Hey Joe!

This is actually not to show the user how to set the working directory, this is to make the vignette build. For whatever reason, in between headers, the script looses the working directory. If these lines were taken out, the script wouldn't run. If there's a solution to this, please let me know!

Reply:
Thanks for the explanation @nkorinek Can you provide a specific error message that occurs or an example of how you know that the script doesn't run without these lines? When I remove these extra code lines to set the working directory, the script does run and the docs do build. Thanks.

Reply:
@jlpalomino weird! On my computer, if I take out those lines, I get this error:

```
ValueError: The output directory path that you provided does not exist
```

I'm fairly sure @lwasser has experienced the error that I am getting as well. If we can run the script without setting the directory for every section that would obviously be ideal!

Reply:
That is strange - maybe an OS issue? I believe you use Windows, Leah uses Mac, and I use Linux.  OK, I will investigate this further, but we can go ahead and say that you have resolved my comments for this PR.

Reply:
weird @jlpalomino so when you build the vignette locally, it builds fine without having to reset the working dir in each section? i had the same issue that @nkorinek but i also havent tried it in a while. it would be good to understand what is happening!

Reply:
@lwasser Yes, on my computer, the script runs and the docs build without having to reset the working directory in each section.  I will set up the earthpy-dev environment on my Windows VM and play around to see if I can figure out what is going on.

Reply:
Just as an update I tried it again on my pc without resetting the directories and it still had the same issues.

Reply:
Update - I tested the vignette script and the doc build in earthpy-dev environment on my Windows VM, and both run successfully. I also do not have to set the PROJ_LIB directory. @lwasser I was not able to replicate the error - should we ask another team member with Windows to test this? @nkorinek what version of Windows are you running?

Reply:
@jlpalomino if @nkorinek submits this to a branch, we can test the build on read the docs. at the end of the day that is the build that matters. how about that? i don't think we need to ask anyone else to test it at this point. Let's 
1. submit the PR to a different branch that we can merge -- maybe vignette3. 
2. setup RTD to build that branch

If it builds, we are ok. especially if it works for you!